Library/Homebrew/cmd/update.sh

```diff
@@ -334,9 +334,26 @@ EOS
     UPSTREAM_BRANCH="$(upstream_branch)"
     # the refspec ensures that the default upstream branch gets updated
     (
-      git fetch "${QUIET_ARGS[@]}" origin \
-        "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" || \
-          odie "Fetching $DIR failed!"
+      UPSTREAM_REPOSITORY_URL="$(git config remote.origin.url)"
+      if [[ "$UPSTREAM_REPOSITORY_URL" = "https://github.com/"* ]]
+      then
+        UPSTREAM_REPOSITORY="$(echo "$UPSTREAM_REPOSITORY_URL" | \
+          sed -e "s|^https://github.com/||" -e "s|.git$||")"
```

Comment:
The Bash way of doing this would be to remove the `https://github.com/` prefix and then to remove the `.git` suffix. The parameter substitution used for this cannot be nested, so the code becomes slightly more repetitive as it requires two separate statements:

```shell
UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY_URL#https://github.com/}"
UPSTREAM_REPOSITORY="${UPSTREAM_REPOSITORY%.git}"
```